manifests/istio-telemetry/kiali/templates/kiali-operator-deployment.yaml

```diff
@@ -0,0 +1,99 @@
+{{- if .Values.kiali.operator.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kiali-operator
+  # IT IS DIRECTLY IN THE CONTROL NAMESPACE FOR NOW
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: kiali-operator
+    release: {{ .Release.Name }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kiali-operator
+  template:
+    metadata:
+      name: kiali-operator
+      labels:
+        app: kiali-operator
+        release: {{ .Release.Name }}
+        version: "{{ .Values.kiali.operator.tag }}"
+      annotations:
+        sidecar.istio.io/inject: "false"
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "8383"
+        {{- if .Values.kiali.operator.podAnnotations }}
+{{ toYaml .Values.kiali.operator.podAnnotations | indent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: kiali-operator-service-account
+{{- if .Values.global.priorityClassName }}
+      priorityClassName: "{{ .Values.global.priorityClassName }}"
+{{- end }}
+{{- if .Values.kiali.operator.tolerations }}
+      tolerations:
+{{ toYaml .Values.kiali.operator.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.kiali.operator.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.kiali.operator.nodeSelector | indent 8 }}
+{{- end }}
+      containers:
+      - name: ansible
+        command:
+        - /usr/local/bin/ao-logs
+        - /tmp/ansible-operator/runner
+        - stdout
+        image: "{{ .Values.kiali.operator.hub }}/{{ .Values.kiali.operator.image }}:{{ .Values.kiali.operator.tag }}"
+{{- if .Values.global.imagePullPolicy }}
+        imagePullPolicy: {{ .Values.global.imagePullPolicy }}
+{{- end }}
+        volumeMounts:
+        - mountPath: /tmp/ansible-operator/runner
+          name: runner
+          readOnly: true
+        resources:
+{{- if .Values.kiali.operator.resources }}
+{{ toYaml .Values.kiali.operator.resources | indent 10 }}
+{{- else }}
+{{ toYaml .Values.global.defaultResources | indent 10 }}
+{{- end }}
+      - name: operator
+        image: "{{ .Values.kiali.operator.hub }}/{{ .Values.kiali.operator.image }}:{{ .Values.kiali.operator.tag }}"
+        imagePullPolicy: {{ .Values.global.imagePullPolicy | default "Always" }}
+        volumeMounts:
+        - mountPath: /tmp/ansible-operator/runner
+          name: runner
+        env:
+        # For now, we assume this Kiali operator installed by Istio will only watch the namespace the operator is in.
```

Comment:
seems like a bad assumption. Is it possible, in the next 4 weeks, to remove this assumption?

Reply:
Yes.

In fact, how this PR is currently implemented here isn't how Kubernetes Operators typically are supposed to work and not how Kiali Operator works in others contexts (e.g. Maistra). But the reason why this assumption is here was to make the PR implementation easier. The alternative would be for this PR to touch more global files such that istioctl creates a separate namespace into which the Kiali Operator is installed, and then the Operator can watch all namespaces for Kiali CRs - this is how multi-tenancy support is handled by Kiali - it is the typical way Kubernetes Operators work.

Again, we could opt to rip out this Kiali Operator installation helm chart entirely in istioctl and require the user to install operators (such as Kiali Operator) as a pre-install step for istioctl.

Reply:
I realize we could opt to rip out the install, and provide a pre-install step. This PR seems to indicate you wish to maintain Kiali install within istioctl. I am good with this approach - although again, you will need to limit the config options to those most mandatory. Another approach would be to limit to those absolutely mandatory and provide a CR override as you have suggested. I think this code could be modified to do that.